src/jvmTest/kotlin/bdd/SpaceEngineersCucumberTest.kt

```diff
@@ -85,19 +97,22 @@ class SpaceEngineersCucumberTest {
     @Given("I load scenario {string}.")
     fun i_load_scenario(scenarioId: String) {
         environment?.let { wrapper ->
-            check(wrapper.controller is WorldController)
-            (wrapper.controller as WorldController).load(File("$SCENARIO_DIR$scenarioId").absolutePath)
+            wrapper.session.loadScenario(File("$SCENARIO_DIR$scenarioId").absolutePath)
         }
-        sleep(500)
         // All blocks are new for the first request.
-        environment.observeNewBlocks()
+        environment.observer.observeNewBlocks().let {
+            assertTrue(it.allBlocks.isNotEmpty())
+        }
+        environment.observer.observeNewBlocks().let {
+            assertTrue(it.allBlocks.isEmpty())
+        }
```

Comment:
this might be a bit fragile, you can get some new blocks when the character moves or some blocks are created

also for the first call you can get zero blocks if there are none within the range

Reply:
Yeah, I will remove that. I added those checks when I though something is wrong with this part of functionality to check what's going on, but problem was elsewhere anyway.